lib/activity_runtime_api.rb

```diff
@@ -448,4 +451,15 @@ def self.remove_report_embeddable_filters(external_activity)
     filters = template.offerings.map { |offering| offering.report_embeddable_filter }.compact
     filters.each { |filter| filter.clear }
   end
+
+
+  def self.update_external_report(external_activity, report_url)
+    external_report = nil
+    if report_url
+      external_report = ExternalReport.find_by_url(report_url)
```

Comment:
I don't think that this will abort.  `find` and `find_by` have different failure behaviors.  See [`find_by` docs](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find_by)  

I could add a test to verify / assert this, though.